js/app.js

```diff
@@ -521,17 +607,30 @@
 		@returns list
 		@description
 			given two lists of items
-			- *EACH* item in `items` must be an object 
+			- *EACH* item in `items` must be an object
 			that looks like this:
 			{
 				'item': 'eggs',
 				'price': 1.59
 			} (else throw error)
-			- return ONE list that contains items in 
+			- return ONE list that contains items in
 			items1 THEN items in items2 as a single array
 	*/
 
 	// implement function here
+	const combineLists = (items1, items2) => {
+		items1.forEach((curr) => {
+			if (!curr.hasOwnProperty('item') || !curr.hasOwnProperty('price')) {
+				throw new Error('must have item and price');
+			}
+		});
+		items2.forEach((curr) => {
+			if (!curr.hasOwnProperty('item') || !curr.hasOwnProperty('price')) {
+				throw new Error('must have item and price');
+			}
+		});
+		return items1.concat(items2);
+	}
```

Comment:
Since you're an advanced student, I'd like to see if you can think of a way to be DRYer. What if we had _n_ items?